client/src/components/layout/ReviewForm.js

```diff
@@ -0,0 +1,94 @@
+import React from "react"
+import ErrorList from "./ErrorList"
+import translateServerErrors from "../../services/translateServerErrors.js"
+const { useState } = require("react")
+
+const ReviewForm = ({buildId, build, setBuild}) => {
+    const emptyFieldsObject = { rating: null, comment: "" }
+    const [newReview, setNewReview] = useState(emptyFieldsObject)
+    console.log(newReview)
+
+    const handleReviewForm = (event) => {
+        setNewReview(
+            {...newReview,
+            [event.currentTarget.name]: event.currentTarget.value}
+        )
+    }
+    
+    const [reviewErrors, setReviewErrors] = useState({})
+    
+    const validateReviewForm = () => {
+        const requiredFields = [{rating: "Rating"}]
+        
+        let errors = {}
+        for (const fieldObject of requiredFields) {
+            for (const key in fieldObject) {
+                if (!newReview[key]) {
+                    errors = {...errors, [fieldObject[key]]: "is required!"}
+                }
+            }
+        }
+        if (!_.isEmpty(errors)) {
+            setReviewErrors(errors)
+            return false
+        } else {
+            return true
+        }
+    }
+
+    const postReview = async (event) => {
+        event.preventDefault()
+        if (validateReviewForm()) {
+            try {
+                const response = await fetch(`/api/v1/builds/${buildId}/reviews`, {
+                    method: "POST",
+                    headers: new Headers({ "Content-Type": "application/json" }),
+                    body: JSON.stringify(newReview)
+                })
+                if (!response.ok) {
+                    if(response.status === 422) {
+                        const body = await response.json()
+                        const newErrors = translateServerErrors(body.errors)
+                        return setReviewErrors(newErrors)
+                    } else {
+                        const errorMessage = `${response.status} (${response.statusText})`
+                        const error = new Error(errorMessage)
+                        throw(error)
+                    }
+                } else {
+                    const body = await response.json()
+                    const updatedReviewArray = build.reviews.concat(body.reviewToAdd)
+                    setNewReview(emptyFieldsObject)
+                    setReviewErrors({})
+                    setBuild({...build, reviews: updatedReviewArray})
+                }
+            } catch(error) {
```

Comment:
fix the formatting on lines 49 above and 65
https://github.com/airbnb/javascript?tab=readme-ov-file#whitespace--around-keywords

Reply:
good